readlimit_test.go

```diff
@@ -65,113 +71,64 @@ func TestMessage_canRead(t *testing.T) {
 }
 
 func TestMessage_ResetReadLimit(t *testing.T) {
-	{
-		m := &Message{TraverseLimit: 42}
-		t.Log("   m := &Message{TraverseLimit: 42}")
-		ok := m.canRead(42)
-		t.Logf("   m.canRead(42) -> %t", ok)
-		m.ResetReadLimit(8)
-		t.Log("   m.ResetReadLimit(8)")
-		if m.canRead(8) {
-			t.Log("   m.canRead(8) -> true")
-		} else {
-			t.Error("!! m.canRead(8) -> false; want true")
-		}
-	}
-	t.Log()
-	{
-		m := &Message{TraverseLimit: 42}
-		t.Log("   m := &Message{TraverseLimit: 42}")
-		ok := m.canRead(40)
-		t.Logf("   m.canRead(40) -> %t", ok)
-		m.ResetReadLimit(8)
-		t.Log("   m.ResetReadLimit(8)")
-		if m.canRead(9) {
-			t.Error("!! m.canRead(9) -> true; want false")
-		} else {
-			t.Log("   m.canRead(9) -> false")
-		}
-	}
-	t.Log()
-	{
-		m := &Message{TraverseLimit: 42}
-		t.Log("   m := &Message{TraverseLimit: 42}")
-		ok := m.canRead(42)
-		t.Logf("   m.canRead(42) -> %t", ok)
-		m.ResetReadLimit(8)
-		t.Log("   m.ResetReadLimit(8)")
-		if m.canRead(8) {
-			t.Log("   m.canRead(8) -> true")
-		} else {
-			t.Error("!! m.canRead(8) -> false; want true")
-		}
-	}
-	t.Log()
-	{
-		m := &Message{TraverseLimit: 42}
-		t.Log("   m := &Message{TraverseLimit: 42}")
-		ok := m.canRead(40)
-		t.Logf("   m.canRead(40) -> %t", ok)
-		m.ResetReadLimit(8)
-		t.Log("   m.ResetReadLimit(8)")
-		if m.canRead(9) {
-			t.Error("!! m.canRead(9) -> true; want false")
-		} else {
-			t.Log("   m.canRead(9) -> false")
-		}
-	}
-	t.Log()
-	{
-		m := new(Message)
-		t.Log("   m := new(Message)")
-		m.ResetReadLimit(0)
-		t.Log("   m.ResetReadLimit(0)")
-		if !m.canRead(0) {
-			t.Error("!! m.canRead(0) -> false; want true")
-		} else {
-			t.Log("   m.canRead(0) -> true")
-		}
-	}
-	t.Log()
-	{
-		m := new(Message)
-		t.Log("   m := new(Message)")
-		m.ResetReadLimit(0)
-		t.Log("   m.ResetReadLimit(0)")
-		if m.canRead(1) {
-			t.Error("!! m.canRead(1) -> true; want false")
-		} else {
-			t.Log("   m.canRead(1) -> false")
-		}
+	t.Parallel()
+	t.Helper()
```

Comment:
Why are we using Helper() here?

Reply:
All the tests are being run as subtests (see line 97).  This helps keep the output tidy.

Reply:
Fair enough. lgtm, but you've got this marked as a draft.